spec/index.bs

```diff
@@ -15836,11 +15836,40 @@ outputs the fragment depth and stencil data to be written into the depth/stencil
 
 ### Output Merging ### {#output-merging}
 
-<p class="note editorial"><span class=marker>Editorial note:</span> fill out this section
+Output merging is a fixed-function stage of the render [=pipeline=] that writes the fragment color values into the
+render pass color attachments.
 
-The depth input to this stage, if any, is clamped to the current
-{{RenderState/[[viewport]]}} depth range
-(regardless of whether the fragment shader stage writes the `frag_depth` builtin).
+<div algorithm data-timeline=queue>
+    <dfn abstract-op>process color attachments</dfn>(fragment, descriptor, state)
+
+    **Arguments:**
+
+    - |fragment|: The [=Fragment=], produced by [[#fragment-processing]].
+    - |descriptor|: The descriptor of type {{GPURenderPipelineDescriptor}}.
+    - |state|: The active [=RenderState=].
+
+    1. If |fragment|.[=Fragment/depthPassed=] is `false` or |fragment|.[=Fragment/stencilPassed=] is `false`, return.
+
+    1. For each |attachment| of |state|.{{RenderState/[[colorAttachments]]}}:
+        1. Let |color| be the value from |fragment|.[=Fragment/colors=] that corresponds with |attachment|.
+        1. Let |targetDesc| be the |descriptor|.{{GPURenderPipelineDescriptor/fragment}}.{{GPUFragmentState/targets}}
+            entry that corresponds with |attachment|.
+        
+        1. If |targetDesc|.{{GPUColorTargetState/blend}} is [=map/exist|provided=]:
+            1. Let |attachmentColor| be the value of |attachment| at |fragment|.[=Fragment/destination=].
+            1. Let |colorBlend| be |targetDesc|.{{GPUColorTargetState/blend}}.{{GPUBlendState/color}}.
+            1. Let |alphaBlend| be |targetDesc|.{{GPUColorTargetState/blend}}.{{GPUBlendState/alpha}}.
+            1. Set the RGB components of |color| to the value computed by performing the operation described by
+                |colorBlend|.{{GPUBlendComponent/operation}} with |colorBlend|.{{GPUBlendComponent/srcFactor}} applied
```

Comment:
We should have something for dual source blend here, I don't see it.

Reply:
Good point. To fix this is actually _simplified_ the algorithm to no longer refer to the explicit attachment that the factors were used with in favor of simply letting the factor descriptions speak for themselves. Even if it's slightly handwavy I think this should allow it to encompass dual source blending now.

Reply:
Didn't see what you had before but this is good. Better to avoid specifying the same things twice.